CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+# Coding Style Guide
+
+The Scala.js project has pretty drastic rules in terms of coding style.
```

Comment:
I would try to reformulate this in a more friendly manner.. something along the lines of

The scala.js project tries to adhere to a single coding standard, to make it easier for everyone to contribute.
This document is part of an ongoing effort to document these best practices and contributions are highly encouraged to follow them.